plugins/cookie_manager/lib/src/cookie_mgr.dart

```diff
@@ -24,9 +24,19 @@ class CookieManager extends Interceptor {
   @override
   void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
     cookieJar.loadForRequest(options.uri).then((cookies) {
-      final cookie = getCookies(cookies);
-      if (cookie.isNotEmpty) {
-        options.headers[HttpHeaders.cookieHeader] = cookie;
+      final newCookie = options.headers[HttpHeaders.cookieHeader] as String?;
+      if (newCookie != null && newCookie.isNotEmpty) {
+        final newCookies = newCookie
+            .split(';')
+            .map((cookie) => Cookie.fromSetCookieValue(cookie))
+            .toList();
+        options.headers[HttpHeaders.cookieHeader] =
+            getCookies(<Cookie>[...cookies, ...newCookies]);
+      } else {
+        final cookie = getCookies(cookies);
+        if (cookie.isNotEmpty) {
+          options.headers[HttpHeaders.cookieHeader] = cookie;
+        }
       }
```

Comment:
The whole flow seems can be simplified by:

```dart
options.headers[HttpHeaders.cookieHeader] = getCookies([
  ...cookies,
  ...?options.headers[HttpHeaders.cookieHeader]
      ?.split(';')
      .map((c) => Cookie.fromSetCookieValue(c)),
]);
```

Reply:
> The whole flow seems can be simplified by:
> 
> ```dart
> options.headers[HttpHeaders.cookieHeader] = getCookies([
>   ...cookies,
>   ...?options.headers[HttpHeaders.cookieHeader]
>       ?.split(';')
>       .map((c) => Cookie.fromSetCookieValue(c)),
> ]);
> ```

looks good